encoding/thrift/thriftrw-plugin-yarpc/fx.go

```diff
@@ -45,8 +60,9 @@ package <$pkgname>
 // 		newHandler,
 // 	)
 func Client(name string, opts ...<$thrift>.ClientOption) interface{} {
-	return func(d *<$yarpc>.Dispatcher) <$client>.Interface {
-		return <$client>.New(d.ClientConfig(name), opts...)
+	return func(p Params) Result {
+		client := <$client>.New(p.Provider.ClientConfig(name), opts...)
+		return Result { Client: client }
```

Comment:
formatting: This isn't how gofmt-formatted code looks. Literals are always like, `Result{Client: client}`